### DISCUSSION TITLE
How should I setup tests?

### DISCUSSION THREAD

**Discussion by phantomdenkali** [thunderclient/thunder-client-support#1268] [Q&A] [answered]
So, I am looking at setting up tests with Thunder. I read the[ testing doc](https://github.com/rangav/thunder-client-support/blob/master/docs/testing.md), but still have a few questions around how I should setup my tests:

Say, I have an API that takes a POST request to add a user exposed at, http://localhost:8000/api/v1/user, there are various tests I want to check for:

1. Successful add user (with well formed json in post)
2.  Re-add the user should correctly fail with duplicate id
3.  Malformed/incorrect json being posted should fail (for e.g. check username length)
...

Basically, I will have different json docs, ordering and looking at different error codes. I get how I can look into the error codes and possibly error codes in the test configuration. However, I do need the tests to take different json docs, and possibly have a specific order. Do I need to setup these tests as separate requests or is there a way to do that as part of the test screen? 

Any help appreciated!

**Comment by rangav** [comment, marked as answer]
Hi @phantomdenkali 

there are 2 options

1. you have to add different requests for each input json body
2. You have a environment variable in place of json in request body. Then in run collection you can input data file with different values for body as array. Then run that request 3 iterations


You can choose any based on your preference.

more advanced using scripting.
- you can use pre-request or post request script run a request in a loop for different inputs
